src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -0,0 +1,24 @@
+module Algebra.Graph.Bipartite.AdjacencyMap (
+    AdjacencyMap, leftToRight, rightToLeft,
+    consistent,
+    ) where
+
+import qualified Data.Map.Strict as Map
+import qualified Data.Set        as Set
+
+data AdjacencyMap a b = BAM {
+    leftToRight :: Map.Map a (Set.Set b),
+    rightToLeft :: Map.Map b (Set.Set a)
+} deriving Eq
+
+consistent :: (Ord a, Ord b) => AdjacencyMap a b -> Bool
+consistent (BAM lr rl) = oneWayConstistent lr rl && oneWayConstistent rl lr
```

Comment:
The current invariant seems too complicated. Couldn't we just do this?

```haskell
consistent (BAM lr rl) = internalEdgeList lr == sort (map swap $ internalList rl)
```

Reply:
@snowleopard, good point, it looks much easier to understand this way!